### PR TITLE
fix: check if port in use before showing success message on local publishing

### LINKED ISSUE #3728: Test in Emulator becomes active before bot is actually ready to accept connections
## Describe the bug

On OS X, the Test In Emulator button becomes active before the bot is ready to accept connections.  This causes the emulator to throw a Directline error.

## Version

Bot Framework Composer 
                Version:  1.0.2
                Electron: 8.2.4
                Chrome: 80.0.3987.165
                NodeJS: 12.13.0
                V8: 8.0.426.27-electron.0
              
## Browser

- [X] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [X] macOS 10.15.6 (19G73)
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a bot in Composer
2. Start the emulator
3. Press Restart Bot
4. Wait for Test in Emulator to be show
5. Press Test in Emulator after it becomes active quickly
6. Observe Directline error in emulator

## Expected behavior

Expect the emulator to be able to connect to the bot as the Test in Emulator button should not be active until the bot is ready to accept connections.

## Screenshots

Screen Shot 2020-07-28 at 5 40 44 AM

### PR DESCRIPTION
**Pull request by zidaneymar**
## Description

Before showing "Test In Emulator", check if the port in use already

## Task Item
fixes #3728

### CODE REVIEW

File: extensions/localPublish/package.json [hunk 1]
```diff
@@ -12,12 +12,14 @@
   "dependencies": {
     "@bfc/extension": "file:../../Composer/packages/extension",
     "@botframework-composer/types": "latest",
+    "@types/tcp-port-used": "1.0.0",
```
Comment (a-b-r-o-w-n) [inline extensions/localPublish/package.json:15]: Is this a production dependency?

Comment (zidaneymar) [reply extensions/localPublish/package.json:15]: > Is this a production dependency?

yes, this is defined here: https://www.npmjs.com/package/@types/tcp-port-used

Comment (cwhitten) [reply extensions/localPublish/package.json:15]: @zidaneymar it looks to be exporting the types alone. Why can this be defined as a `devDependency` ?

Comment (zidaneymar) [reply extensions/localPublish/package.json:15]: > @zidaneymar it looks to be exporting the types alone. Why can't this be defined as a `devDependency` ?

sorry I will fix this

**Comment by cwhitten** [comment]
@zidaneymar please fix tests

**Comment by zidaneymar** [comment]
> @zidaneymar please fix tests

I have fixed the ut and resolved the conflicts.

**Review by cwhitten — APPROVED** [review]

**Pull request merged**